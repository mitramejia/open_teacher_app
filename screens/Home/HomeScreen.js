import React from 'react';
import { Button, Text, Content } from 'native-base';
import StudentHomeScreen from '../Student/Home/StudentHomeScreen';
import LoginScreen from '../Login/LoginScreen';

export default class HomeScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      userIsLoggedIn: false,
    };
  }

  static navigationOptions = {
    header: null,
  };

  render() {
    if (this.state.userIsLoggedIn) {
      return <StudentHomeScreen navigation={this.props.navigation} />;
    } else {
      return <LoginScreen navigation={this.props.navigation} />;
    }
  }
}
