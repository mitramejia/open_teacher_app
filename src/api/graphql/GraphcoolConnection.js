// =============================================================
//
// Name: GraphcoolConnection
// -> Description:
//
// Author: mitramejia
// Created at: 8/21/17
//
// =============================================================

import { ApolloClient, createNetworkInterface } from 'apollo-client';
import { GRAPHCOOL_SIMPLE_API, GRAPHCOOL_WEB_SOCKET_CLIENT } from 'react-native-dotenv';
import { addGraphQLSubscriptions, SubscriptionClient } from 'subscriptions-transport-ws';
import { AsyncStorage } from 'react-native';

const networkInterface = createNetworkInterface({ uri: GRAPHCOOL_SIMPLE_API });

// Use middleware to manage auth tokens
networkInterface.use([
  {
    applyMiddleware(req, next) {
      if (!req.options.headers) {
        req.options.headers = {}; // Create the header object if needed.
      }

      AsyncStorage.getItem('token').then(
        encodedToken => {
          req.options.headers['authorization'] = `Bearer ${encodedToken}`;
          next();
        },
        failure => {
          console.error('ERROR: no token', failure);
          next();
        }
      );
    },
  },
]);

// Create WebSocket client
const wsClient = new SubscriptionClient(GRAPHCOOL_WEB_SOCKET_CLIENT, {
  reconnect: true,
});
// Extend the network interface with the WebSocket
const networkInterfaceWithSubscriptions = addGraphQLSubscriptions(networkInterface, wsClient);

const client = new ApolloClient({
  networkInterface: networkInterfaceWithSubscriptions,
  dataIdFromObject: o => o.id,
});

export default client;
