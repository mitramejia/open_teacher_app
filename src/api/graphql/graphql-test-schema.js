// =============================================================
//
// Name: graphql-test-schema
// -> Description:
//
// Author: mitramejia
// Created at: 8/28/17
//
// =============================================================

import { makeExecutableSchema, addMockFunctionsToSchema } from 'graphql-tools';
import { mockNetworkInterfaceWithSchema } from 'apollo-test-utils';
import { ApolloClient } from 'apollo-client';

export function mockApolloClient() {
  const schema = makeExecutableSchema({ typeDefs });
  addMockFunctionsToSchema({ schema });
  const mockNetworkInterface = mockNetworkInterfaceWithSchema({ schema });
  return new ApolloClient({
    networkInterface: mockNetworkInterface,
  });
}

export const typeDefs = `
      type User {
        id: Int
        firstName: String
        lastName: String
        email: String
      }

      type Query {
        user: User
      }
    `;
