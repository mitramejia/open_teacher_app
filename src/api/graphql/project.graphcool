# project: cj5zswpmnfr6201049umywa1m
# version: 48

type Post implements Node {
  description: String!
  imageUrl: String!
  createdAt: DateTime!
  id: ID! @isUnique
  updatedAt: DateTime!
}

type File implements Node {
  contentType: String!
  createdAt: DateTime!
  id: ID! @isUnique
  name: String!
  secret: String! @isUnique
  size: Int!
  updatedAt: DateTime!
  url: String! @isUnique
}

type Tutor implements Node {
  createdAt: DateTime!
  id: ID! @isUnique
  updatedAt: DateTime!
  students: [Student!]! @relation(name: "TutorOnStudent")
  description: String
  diplomaUrl: String!
  lessons: [Lesson!]! @relation(name: "TutorOnLesson")
  locations: [String!]!
  subjects: [Subject!]! @relation(name: "SubjectOnTutor")
  user: User! @relation(name: "UserOnTutor")
}

type User implements Node {
  createdAt: DateTime!
  firstName: String!
  imageUrl: String
  lastName: String!
  id: ID! @isUnique
  updatedAt: DateTime!
  email: String @isUnique
  isTutor: Boolean! @defaultValue(value: false)
  password: String
  student: Student @relation(name: "UserOnStudent")
  tutor: Tutor @relation(name: "UserOnTutor")
}

type Lesson implements Node {
  cost: Float!
  createdAt: DateTime!
  duration: Int!
  id: ID! @isUnique
  student: Student @relation(name: "StudentOnLesson")
  updatedAt: DateTime!
  user: Tutor @relation(name: "TutorOnLesson")
}

type Student implements Node {
  createdAt: DateTime!
  id: ID! @isUnique
  lessons: [Lesson!]! @relation(name: "StudentOnLesson")
  subjects: [Subject!]! @relation(name: "SubjectOnStudent")
  tutor: Tutor @relation(name: "TutorOnStudent")
  updatedAt: DateTime!
  user: User! @relation(name: "UserOnStudent")
}

type Subject implements Node {
  createdAt: DateTime!
  id: ID! @isUnique
  name: String!
  students: [Student!]! @relation(name: "SubjectOnStudent")
  tutors: [Tutor!]! @relation(name: "SubjectOnTutor")
  updatedAt: DateTime!
}