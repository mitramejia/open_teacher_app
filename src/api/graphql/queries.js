// =============================================================
//
// Name: model
// -> Description:
//
// Author: mitramejia
// Created at: 8/25/17
//
// =============================================================

import { gql } from 'react-apollo';

export const createAccount = gql`
  mutation($firstName: String!, $lastName: String!, $email: String!, $password: String!) {
    createUser(
      authProvider: { email: { email: $email, password: $password } }
      firstName: $firstName
      lastName: $lastName
    ) {
      id
      firstName
      lastName
      email
      password
    }
  }
`;

export const userQuery = gql`
  query {
    user {
      id
      firstName
      lastName
      email
    }
  }
`;

export const signinUser = gql`
  mutation($email: String!, $password: String!) {
    signinUser(email: { email: $email, password: $password }) {
      token
    }
  }
`;

export const allSubjectsQuery = gql`
  query {
    allSubjects {
      id
      name
    }
  }
`;
