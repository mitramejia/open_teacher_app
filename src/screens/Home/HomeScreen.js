import React from 'react';
import { AsyncStorage } from 'react-native';
import { Button, Content } from 'native-base';
import PropTypes from 'prop-types';
import SearchTutorScreen from '../Student/SearchTutor/SearchTutorScreen';
import LoginScreen from '../Login/LoginScreen';
import TutorHomeScreen from '../Tutor/Home/TutorHomeScreen';

export class HomeScreen extends React.Component {
  state = {
    user: null,
  };

  static navigationOptions = {
    header: null,
  };

  _logout = async () => {
    await AsyncStorage.removeItem('token');
    this.props.client.resetStore();
  };

  render() {
    if (this.props.data.user) {
      if (this.props.data.user.isTutor) {
        return <TutorHomeScreen user={this.props.data.user} navigation={this.props.navigation} />;
      } else {
        return <SearchTutorScreen user={this.props.data.user} navigation={this.props.navigation} />;
      }
    } else {
      return <LoginScreen client={this.props.client} navigation={this.props.navigation} />;
    }
  }
}

HomeScreen.propTypes = {
  state: PropTypes.shape({
    user: PropTypes.any.isRequired,
  }),
  navigation: PropTypes.object.isRequired,
};

export default HomeScreen;
