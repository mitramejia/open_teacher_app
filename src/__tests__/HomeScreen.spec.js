import React from 'react';
import HomeScreen from '../screens/Home/HomeScreen';
import renderer from 'react-test-renderer';
import { ApolloProvider } from 'react-apollo';
import { mockApolloClient } from '../api/graphql/graphql-test-schema';

it('renders the Home Screen', () => {
  const navigation = {};
  const tree = renderer
    .create(
      <ApolloProvider client={mockApolloClient()}>
        <HomeScreen navigation={navigation} />
      </ApolloProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
