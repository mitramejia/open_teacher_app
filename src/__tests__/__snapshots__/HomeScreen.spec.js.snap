// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Home Screen 1`] = `<ExponentAppLoading />`;
